frontend/public/components/replication-controller.jsx

```diff
@@ -27,40 +26,67 @@ import {
-  // t('public~Cancel rollout')
-  labelKey: 'public~Cancel rollout',
-  hidden: !_.includes(
-    ['New', 'Pending', 'Running'],
-    obj?.metadata?.annotations?.['openshift.io/deployment.phase'],
-  ),
-  callback: () =>
-    confirmModal({
-      title: i18next.t('public~Cancel rollout'),
-      message: i18next.t('public~Are you sure you want to cancel this rollout?'),
-      btnText: i18next.t('public~Yes, cancel'),
-      cancelText: i18next.t("public~No, don't cancel"),
-      executeFn: () =>
-        k8sPatch(kind, obj, [
-          {
-            op: 'add',
-            path: '/metadata/annotations/openshift.io~1deployment.cancelled',
-            value: 'true',
-          },
-          {
-            op: 'add',
-            path: '/metadata/annotations/openshift.io~1deployment.status-reason',
-            value: 'cancelled by the user',
-          },
-        ]),
-    }),
-  accessReview: asAccessReview(kind, obj, 'patch'),
-});
+const INACTIVE_STATUSES = ['New', 'Pending', 'Running'];
 
-const menuActions = [CancelAction, ...replicaSetMenuActions];
+const CancelAction = (kind, obj) => {
+  const deploymentPhase = obj?.metadata?.annotations?.['openshift.io/deployment.phase'];
+  return {
+    // t('public~Cancel rollout')
+    labelKey: 'public~Cancel rollout',
+    hidden: !deploymentPhase || !INACTIVE_STATUSES.includes(deploymentPhase),
+    callback: () =>
+      confirmModal({
+        title: i18next.t('public~Cancel rollout'),
+        message: i18next.t('public~Are you sure you want to cancel this rollout?'),
+        btnText: i18next.t('public~Yes, cancel'),
+        cancelText: i18next.t("public~No, don't cancel"),
+        executeFn: () =>
+          k8sPatch(kind, obj, [
+            {
+              op: 'add',
+              path: '/metadata/annotations/openshift.io~1deployment.cancelled',
+              value: 'true',
+            },
+            {
+              op: 'add',
+              path: '/metadata/annotations/openshift.io~1deployment.status-reason',
+              value: 'cancelled by the user',
+            },
+          ]),
+      }),
+    accessReview: asAccessReview(kind, obj, 'patch'),
+  };
+};
+
+const RollbackAction = (kind, obj) => {
+  const deploymentPhase = obj?.metadata?.annotations?.['openshift.io/deployment.phase'];
+  const dcName = getDeploymentConfigNameForRC(obj);
+  return {
+    // t('public~Rollback')
+    labelKey: 'public~Rollback',
+    // disabled if the DC is not Active
+    isDisabled: INACTIVE_STATUSES.includes(deploymentPhase),
+    // Hidden if RC is active or does not belong to a deployment config
+    hidden: !deploymentPhase || obj?.status?.replicas > 0 || !dcName,
+    callback: () =>
+      rollbackRCModal({
+        resourceKind: kind,
+        resource: obj,
+      }),
+    accessReview: {
+      group: DeploymentConfigModel.apiGroup,
+      resource: DeploymentConfigModel.plural,
+      name: dcName,
+      namespace: obj?.metadata?.namespace,
+    },
```

Comment:
Sorry, I forgot to specify the verb in my example, which is needed :(

```suggestion
    accessReview: {
      group: DeploymentConfigModel.apiGroup,
      resource: DeploymentConfigModel.plural,
      name: dcName,
      namespace: obj?.metadata?.namespace,
      verb: 'update',
    },
```

Reply:
thanks! fixed.